piwebapi_samples/Data Analysis/PIUploadUtility/PIUploadUtility/PIUploadUtility/Program.cs

```diff
@@ -0,0 +1,243 @@
+                return response["WebId"].ToString();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.InnerException.Message);
+            }
+
+            return null;
+        }
+
+        static void CreateDatabase(XmlDocument doc, string assetserver)
+        {
+            string serverPath = "\\\\" + assetserver;
+            string assetserverWebID = GetWebIDByPath(serverPath, "assetservers");
+
+            string createDBQuery = "assetservers/" + assetserverWebID + "/assetdatabases";
+
+            string databaseName = config["AF_DATABASE_NAME"].ToString();
+
+            Object payload = new 
+            {
+                 Name = databaseName,
+                 Description = "Example for Building Data"  
+            };
+
+            string request_body = JsonConvert.SerializeObject(payload);
+            
+            try
+            {
+                client.PostRequest(createDBQuery, request_body);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.InnerException.Message);
+            }
+            
+            string databasePath = serverPath + "\\" + databaseName;
+            string databaseWebID = GetWebIDByPath(databasePath, "assetdatabases");
+            string importQuery = "assetdatabases/" + databaseWebID + "/import";
+
+            try
+            {
+                client.PostRequest(importQuery, doc.InnerXml.ToString(), true);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.InnerException.Message);
+            }
+        }
+
+        static void CreatePIPoint(string dataserver, string tagDefinitionLocation)
+        {
+            string path = "\\\\PIServers[" + dataserver + "]";
+            string dataserverWebID = GetWebIDByPath(path, "dataservers");
+            string createPIPointQuery = "dataservers/" + dataserverWebID + "/points";
+            
+            var tagDefinitions = File.ReadLines(tagDefinitionLocation);
+            string name, pointType, pointClass;
+
+            foreach (string tagDefinition in tagDefinitions)
+            {
+                string[] split = tagDefinition.Split(',');
+                name = split[0];
+                pointType = split[1];
+                pointClass = split[2];
+
+                Object payload = new
+                {
+                    Name = name,
+                    PointType = pointType,
+                    PointClass = pointClass
+                };
+
+                string request_body = JsonConvert.SerializeObject(payload);
+
+                try
+                {
+                    client.PostRequest(createPIPointQuery, request_body);
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.InnerException.Message);
+                }
+            }
+        }
+
+        static bool IsTagExist(string dataserver)
```

Comment:
`DoesTagExist`? `IsTagFound`? Grammar seems off.

Reply:
Wow never noticed that. Will fix.